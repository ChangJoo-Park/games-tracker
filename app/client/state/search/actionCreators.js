import {
  RESULTS_REQUESTED,
  RESULTS_RESOLVED,
  RESULTS_REJECTED,
  RESULTS_CLEARED
} from "./actionTypes";

export const requestResults = () => ({
  type: RESULTS_REQUESTED
});

export const resolveResults = results => ({
  type: RESULTS_RESOLVED,
  payload: {
    results
  }
});

export const rejectResults = () => ({
  type: RESULTS_REJECTED
});

export const clearResults = () => ({
  type: RESULTS_CLEARED
});
