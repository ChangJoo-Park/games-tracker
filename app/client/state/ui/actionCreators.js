import { MODAL_OPENED, MODAL_CLOSED } from "./actionTypes";

export const openModal = content => ({
  type: MODAL_OPENED,
  payload: {
    content
  }
});

export const closeModal = () => ({
  type: MODAL_CLOSED
});
