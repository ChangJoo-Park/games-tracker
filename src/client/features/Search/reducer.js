import {
  QUERY_UPDATED,
  RESULTS_REQUESTED,
  RESULTS_RESOLVED,
  RESULTS_REJECTED
} from "./actionTypes";

export const initialState = {
  query: "",
  isSearching: false,
  results: [
    {
      name: "The Witcher 3: Wild Hunt",
      cover:
        "https://images.igdb.com/igdb/image/upload/t_cover_big/tri1c6vbydeosoqajwt1.jpg",
      platforms: ["PC", "PS4", "XONE"]
    },
    {
      name: "The Witcher 2: Assassins of Kings",
      cover:
        "https://images.igdb.com/igdb/image/upload/t_cover_big/mcou7xzxq0qnerehesrh.jpg",
      platforms: ["PC", "MAC", "X360", "LIN"]
    },
    {
      name: "The Witcher",
      cover:
        "https://images.igdb.com/igdb/image/upload/t_cover_big/nrkdzmkevbbevdvm9bxh.jpg",
      platforms: ["PC", "MAC"]
    }
  ]
};

export default (state = initialState, action) => {
  switch (action.type) {
    case QUERY_UPDATED:
      return {
        ...state,
        query: action.payload.query
      };
    case RESULTS_REQUESTED:
      return {
        ...state,
        isSearching: true
      };
    case RESULTS_RESOLVED:
      return {
        ...state,
        isSearching: false,
        results: action.payload.results,
        query: ""
      };
    case RESULTS_REJECTED:
      return {
        ...state,
        isSearching: false,
        results: [],
        query: ""
      };
    default:
      return state;
  }
};
