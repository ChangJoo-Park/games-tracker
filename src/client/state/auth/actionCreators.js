import { USER_SIGNED_IN, USER_SIGNED_OUT } from "./actionTypes";

export const signUserIn = ({ email }) => ({
  type: USER_SIGNED_IN,
  payload: {
    user: {
      email
    }
  }
});

export const signUserOut = () => ({
  type: USER_SIGNED_OUT
});
