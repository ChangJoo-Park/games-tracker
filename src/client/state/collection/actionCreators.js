import { GAMES_RESOLVED, GAME_UPDATE_RESOLVED } from "./actionTypes";

export const resolveGames = games => ({
  type: GAMES_RESOLVED,
  payload: {
    games
  }
});

export const resolveGameUpdate = game => ({
  type: GAME_UPDATE_RESOLVED,
  payload: {
    game
  }
});
