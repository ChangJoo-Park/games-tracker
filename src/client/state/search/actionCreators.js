// @flow
import {
  QUERY_UPDATED,
  RESULTS_REQUESTED,
  RESULTS_RESOLVED,
  RESULTS_REJECTED
} from "./actionTypes";

export type QueryUpdated = {
  type: "Search/QUERY_UPDATED",
  payload: {
    query: string
  }
};

export type UpdateQuery = string => QueryUpdated;

export const updateQuery: UpdateQuery = query => ({
  type: QUERY_UPDATED,
  payload: {
    query
  }
});

export type ResultsRequested = {
  type: "Search/RESULTS_REQUESTED"
};

export const requestResults = (): ResultsRequested => ({
  type: RESULTS_REQUESTED
});

export type ResultsResolved = {
  type: "Search/RESULTS_RESOLVED",
  payload: {
    results: SearchResults
  }
};

export const resolveResults = (results: SearchResults): ResultsResolved => ({
  type: RESULTS_RESOLVED,
  payload: {
    results
  }
});

export type ResultsRejected = {
  type: "Search/RESULTS_REJECTED"
};

export const rejectResults = (): ResultsRejected => ({
  type: RESULTS_REJECTED
});
