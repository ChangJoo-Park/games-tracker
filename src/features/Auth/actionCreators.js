import { USER_SIGNED_IN, USER_SIGNED_OUT } from "./actionTypes";

export const signUserIn = user => ({
  type: USER_SIGNED_IN,
  payload: {
    user
  }
});

export const signUserOut = () => ({
  type: USER_SIGNED_OUT
});
