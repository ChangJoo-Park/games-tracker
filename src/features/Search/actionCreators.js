import {
  UPDATE_QUERY,
  REQUEST_RESULTS,
  RESOLVE_RESULTS,
  REJECT_RESULTS
} from "./actionTypes";

export const updateQuery = query => ({
  type: UPDATE_QUERY,
  payload: {
    query
  }
});

export const requestResults = () => ({
  type: REQUEST_RESULTS
});

export const resolveResults = results => ({
  type: RESOLVE_RESULTS,
  payload: {
    results
  }
});

export const rejectResults = () => ({
  type: REJECT_RESULTS
});
