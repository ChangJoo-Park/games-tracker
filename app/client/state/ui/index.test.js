import React from "react";
import { openModal, closeModal } from "./actionCreators";
import reducer, { initialState } from "./";

describe("Search", () => {
  test("MODAL_OPENED", () => {
    const action = openModal(<p>Modal content</p>);
    const returnedState = reducer(initialState, action);

    expect(returnedState).toMatchSnapshot();
  });

  test("MODAL_CLOSED", () => {
    const action = closeModal();
    const returnedState = reducer(initialState, action);

    expect(returnedState).toMatchSnapshot();
  });
});
