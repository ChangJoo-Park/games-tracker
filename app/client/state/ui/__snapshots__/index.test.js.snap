// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search MODAL_CLOSED 1`] = `
Object {
  "isModalOpen": false,
  "modalContent": null,
}
`;

exports[`Search MODAL_OPENED 1`] = `
Object {
  "isModalOpen": true,
  "modalContent": <p>
    Modal content
  </p>,
}
`;
